CMakeLists.txt

```diff
@@ -604,7 +604,7 @@ else()
     find_package(MatrixClient 0.9.0 REQUIRED)
 endif()
 
-if(VOIP)
+if(VOIP AND NOT HAIKU)
```

Comment:
I suggest you change the VOIP_DEFAULT instead. In theory voip should work on Haiku at least to some extent.

Reply:
Tried putting Haiku in the file there, still picked up "if(VOIP)", another round I checked was adding another check here so it will pick up gstreamer but not the missing gstreamer-gl. 
Thanks for taking the time to look into it, will get back on this in the morning.

Reply:
It just sets the default value, so you need to completely wipe the builddir (or create a new builddir) to test it.

Reply:
Mostly I just remove the cmakecache file, but also did removed the builddir a few times, will check some more today.

Reply:
OK, first attempt today (outcome was the same yesterday), with builddir whiped out.
Using:
```
if(APPLE OR WIN32 OR HAIKU)
    set(VOIP_DEFAULT OFF)
endif()
```
Ends up in: --   No package 'gstreamer-gl-1.0' found

Reply:
Using the next (simular to the current PR) works also:
```
if(VOIP)
if(HAIKU)
    include(FindPkgConfig)
    pkg_check_modules(GSTREAMER REQUIRED IMPORTED_TARGET gstreamer-sdp-1.0>=1.18 gstreamer-webrtc-1.0>=1.18)
else()
    include(FindPkgConfig)
    pkg_check_modules(GSTREAMER REQUIRED IMPORTED_TARGET gstreamer-sdp-1.0>=1.18 gstreamer-webrtc-1.0>=1.18 gstreamer-gl-1.0)
endif()
endif()
```